src/improvements/tasks/OPCMBaseTask.sol

```diff
@@ -0,0 +1,120 @@
+        /// and we don't have to worry about the Claim type as we are not
+        /// calling the interface with that type
+        bytes32 absolutePrestate;
+    }
+
+    /// @notice Call3 struct used in the Multicall3DelegateCall contract
+    struct Call3 {
+        address target;
+        bool allowFailure;
+        bytes callData;
+    }
+
+    /// @notice get the calldata to be executed by safe
+    /// @dev callable only after the build function has been run and the
+    /// calldata has been loaded up to storage
+    /// @return data The calldata to be executed
+    function getCalldata() public view override returns (bytes memory data) {
+        /// get task actions
+        (address[] memory targets,, bytes[] memory arguments) = getTaskActions();
+
+        /// create calls array with targets and arguments
+        Call3[] memory calls = new Call3[](targets.length);
+
+        for (uint256 i; i < calls.length; i++) {
+            require(targets[i] != address(0), "Invalid target for multisig");
+            calls[i] = Call3({target: targets[i], allowFailure: false, callData: arguments[i]});
+        }
+
+        /// generate calldata
+        data = abi.encodeWithSignature("aggregate3((address,bool,bytes)[])", calls);
+    }
+
+    /// @notice get the data to sign by EOA
+    /// @param safe The address of the safe
+    /// @param data The calldata to be executed
+    /// @return The data to sign
+    function getDataToSign(address safe, bytes memory data) public view override returns (bytes memory) {
+        return IGnosisSafe(safe).encodeTransactionData({
+            to: _getMulticallAddress(safe),
+            value: 0,
+            data: data,
+            operation: Enum.Operation.DelegateCall,
+            safeTxGas: 0,
+            baseGas: 0,
+            gasPrice: 0,
+            gasToken: address(0),
+            refundReceiver: address(0),
+            _nonce: _getNonce(safe)
+        });
+    }
+
+    /// @notice execute post-task checks.
+    /// read states that are expected to have changed during the simulate step.
+    /// removes the storage write checks
+    function validate() public view override {
+        require(IGnosisSafe(multisig).nonce() == nonce + 1, "MultisigTask: nonce not incremented");
+
+        Addresses.ChainInfo[] memory chains = addresses.getChains();
+
+        for (uint256 i = 0; i < chains.length; i++) {
+            _validate(chains[i].chainId);
+        }
+    }
+
+    /// @notice get the multicall address for the given safe
+    /// if the safe is the parent multisig, return the delegatecall multicall address
+    /// otherwise if the safe is a child multisig, return the regular multicall address
+    /// @param safe The address of the safe
+    /// @return The address of the multicall
+    function _getMulticallAddress(address safe) internal view returns (address) {
+        return (safe == multisig) ? MULTICALL3_DELEGATECALL_ADDRESS : MULTICALL3_ADDRESS;
+    }
```

Comment:
Can we rename `multisig` to something like `rootMultisig`? I see the natspec for `multisig` is `The address of the multisig for this task` but a task can have 3 involved multisigs in the nested case

Reply:
+1 mentioned here: https://github.com/ethereum-optimism/superchain-ops/pull/599#discussion_r1955283286